php/utils.php

```diff
@@ -884,3 +886,28 @@ function expand_globs( $paths, $flags = GLOB_BRACE ) {
 
 	return array_unique( $expanded );
 }
+
+/**
+ * Get a Phar-safe version of a path.
+ *
+ * For paths inside a Phar, this strips the outer filesystem's location to
+ * reduce the path to what it needs to be within the Phar archive.
+ *
+ * Use the __FILE__ or __DIR__ constants as a starting point.
+ *
+ * @param string $path An absolute path that might be within a Phar.
+ *
+ * @return string A Phar-safe version of the path.
+ */
+function phar_safe_path( $path ) {
+
+	if ( ! inside_phar() ) {
+		return $path;
+	}
+
+	return str_replace(
+		PHAR_STREAM_PREFIX . getcwd() . '/',
```

Comment:
Ah, I understand. So you want to still do the replacement above, but want to catch a constant early on instead of the potentially changing `getcwd()`? Yes, that makes sense.